tests/app/shared/issue/duplicatedIssues/duplicated-issues.component.spec.ts

```diff
@@ -0,0 +1,164 @@
+
+        thisIssue = Issue.createPhaseModerationIssue(ISSUE_PENDING_MODERATION, dummyTeam);
+        firstDummyIssue.duplicateOf = thisIssue.id;
+        secondDummyIssue.duplicateOf = thisIssue.id;
+
+        fixture = TestBed.createComponent(DuplicatedIssuesComponent);
+        component = fixture.componentInstance;
+        component.issue = thisIssue;
+        fixture.detectChanges();
+
+        debugElement = fixture.debugElement;
+        nativeElement = fixture.nativeElement;
+    });
+
+    it('should create a chip for each duplicated issue', () => {
+        // Check that the number of chips is the same as the number of duplicate issues
+        const matChipListLength: number = debugElement.queryAll(By.css('.mat-chip')).length;
+        const duplicateIssueslength: number = getDuplicateIssuesLength();
+        expect(matChipListLength).toEqual(duplicateIssueslength);
+
+        // Check that first chip contains information on first issue in the duplicate issues array
+        const matChipAnchor: HTMLElement = nativeElement.querySelector('a');
+        expect(matChipAnchor.innerText).toEqual(`#${firstDummyIssue.id}`);
+    });
+
+    it('should only allow cancellation of duplicate status if team/tutor response is editable', () => {
+        // Team/tutor response is not editable
+        mockTeamResponseNotEditable(permissionService);
+        mockTutorResponseNotEditable(permissionService);
+        fixture.detectChanges();
+        const cancelIconQuery = debugElement.query(By.css('.mat-icon'));
+        expect(cancelIconQuery).toBeNull();
+
+        // Team response is editable
+        mockTeamResponseEditable(permissionService);
+        mockTutorResponseNotEditable(permissionService);
+        fixture.detectChanges();
+        let cancelIcon: HTMLElement = debugElement.query(By.css('.mat-icon')).nativeElement;
+        expect(cancelIcon).toBeDefined();
+
+        // Tutor response is editable
+        mockTeamResponseNotEditable(permissionService);
+        mockTutorResponseEditable(permissionService);
+        fixture.detectChanges();
+        cancelIcon = debugElement.query(By.css('.mat-icon')).nativeElement;
+        expect(cancelIcon).toBeDefined();
+    });
+
+    it('should remove duplicate status of an duplicated issue once cancel icon is clicked', () => {
+        let updatedFirstDummyIssue: Issue;
+        issueService.updateIssueWithComment.and.callFake((duplicatedIssue: Issue) => {
+            updatedFirstDummyIssue = duplicatedIssue;
+            return of(duplicatedIssue);
+        });
+        spyOn(component, 'removeDuplicateStatus').and.callThrough();
+        mockTeamResponseEditable(permissionService);
+        fixture.detectChanges();
+
+        expect(firstDummyIssue.duplicateOf).toEqual(component.issue.id);
+        cancelDuplicateStatus();
+        expect(component.removeDuplicateStatus).toHaveBeenCalledWith(firstDummyIssue);
```

Comment:
I don't think there's a need to verify that `removeDuplicateStatus` is called.
It's an internal detail within the component.

We can focus on checking the external effects (such as whether the issue's `duplicateOf` property is updated, and whether methods on the `IssueService` dependency are called correctly).

Reply:
Alright, got it!